apps/vmq_server/src/vmq_cluster_node.erl

```diff
@@ -256,12 +272,29 @@ maybe_flush(#state{pending=Pending} = State) ->
             State
     end.
 
+%% In case the receiving node crashes while it still had messages in the receiving tcp buffer
+%% those messages would be lost.
+%% Therefore we introduce a handshake at application level:
+%%  - append a 'tail' to the block of messages that are sent
+%%  - keep a backup copy of the messages in a 'transit' buffer
+%%  - await the block_ack from the receiving node, as it handled the complete block
+%%    (upon receipt of the 'tail')
+%%  - when the block_ack is received, the transit buffer is cleared
+%%    and the next block can be transmitted
+%% This handshake is protected by a timeout which will trigger a reconnect
+%% Note that currently the backup is kept only in memory, a future improvement
+%% would be to persist the backup on disk to also protect against message loss
+%% in case the transmitting node crashes.
```

Comment:
Maybe we should put this in a ticket @ioolkos so it doesn't get lost in the code comments